pkg/auth/iam/iam.go

```diff
@@ -137,6 +128,12 @@ func (filter *Filter) Auth(opts ...FilterOption) restful.FilterFunction {
 			}
 		}
 
+		if filter.options.SubdomainValidationEnabled {
+			if err := matchSubdomainWithNamespace(getHost(req.Request), claims.Namespace, filter.options.SubdomainValidationExcludedNamespaces); err != nil {
```

Comment:
@rexydwybrundy I realized that I cannot do the subdomain validation step in `FilterOption` since I need access to the excluded namespaces.
the env var related to the subdomain has been stated explicitly in the filter init options, so this is the best possible solution that I can think of.